test/cpp/src/702-dram-refresh.cc

```diff
@@ -0,0 +1,94 @@
+#include <catch.hpp>
+#include "mocks.hpp"
+#include "defaults.hpp"
+#include "dram_controller.h"    
+#include "champsim_constants.h"
+#include <algorithm>
+#include <cfenv>
+#include <cmath>
+
+void generate_packet(champsim::channel* channel)
+{
+    auto pkt_type = rand() % 2 ? access_type::LOAD : access_type::WRITE;
+    champsim::channel::request_type r;
+    r.type = pkt_type;
+    uint64_t offset = 0;
+    r.address = 0;
+    r.v_address = 0;
+    offset += LOG2_BLOCK_SIZE;
+    r.address += 0 << offset;
+    offset += champsim::lg2(DRAM_CHANNELS);
+    r.address += rand() % DRAM_BANKS << offset;
+    offset += champsim::lg2(DRAM_BANKS);
+    r.address += 1 << offset;
+    offset += champsim::lg2(DRAM_COLUMNS);
+    r.address += rand() % DRAM_RANKS << offset;
+    offset += champsim::lg2(DRAM_RANKS);
+    r.address += rand() % DRAM_ROWS << offset;
```

Comment:
Avoid creating random packets in these tests. If you need variety, you can use Catch2's `GENERATE` directive, or create them procedurally. Random values in tests could potentially break in ways that are unrelated to a change being made.

Reply:
I have rewritten the test to generate packets based on the current cycle.